src/js/js-loader.ts

```diff
@@ -33,9 +33,9 @@ function jsLoader(source: string, sourceMaps: any): void {
 }
 
 function extractTranslations(loader: TranslateLoaderContext, source: string, sourceMaps: any): void {
-    const options = loaderUtils.parseQuery(loader.query);
+    const options = (loaderUtils as any).getOptions(loader) || {};
```

Comment:
Good point!

You can define it as union type:

```js
query: string | { [name: string] : any }
```

Or probably just assign the type `any` as it can be any object (but no number, but it's not that important)